articles/realtime.md

```diff
@@ -0,0 +1,293 @@
+---
+layout: default
+title: ROS 2.0 in Realtime
+permalink: articles/realtime.html
+abstract: This article describes real-time computing requirements and how ROS 2 will fulfill them.
+published: false
+author: Jackie Kay
+---
+
+* This will become a table of contents (this text will be scraped).
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+Robotic systems need to be responsive. In mission critical applications, a delay of less than a
+millisecond in the system can cause a catastrophic failure. For ROS 2.0 to capture the needs
+of the robotics community, the core software components must not interfere with the requirements of real-time computing.
+
+# Definition of Real-time Computing
+
+Real-time software guarantees correct computation at the correct time. Hard real-time software systems
+have a set of strict deadlines, and missing a deadline is considered a failure. Soft real-time systems
+degrade their quality of service if a deadline is missed.
+
+Real-time computer systems are often associated with low-latency systems. Many applications of real-time
+computing are also low-latency applications (for example, automated piloting systems must be reactive to
+sudden changes in the environment). However, it is
+generally agreed upon that a real-time system is not defined by low latency, but by a deterministic schedule:
+it must be guaranteed that the system finishes a certain task by a certain time. Therefore, it is
+important that the latency in the system be measurable and a maximum allowable latency for tasks be
+set.
+
+A more useful metric in evaluating the "hardness" of a real-time system is jitter, which is defined
+as the variable deviation between a task's deadline and its actual time of completion. A more jittery
+system is less deterministic, less predictable, and less real-time. Though in practice it is impossible
+to completely eliminate jitter from a real-time system, it is a worthy goal to determine a hard upper bound
+for jitter.
```

Comment:
Dissenting opinion: The reference text shouldn't be changing and it is annoying to have to open and read another (potentially large) document to get the meaning. I think it is totally appropriate to quote a snippet of the referenced text or summarize it in this document to make it more self contained. Part of the job of the document is to distill the subject matter into an easy to digest form.